File: src/test/java/com/qomolangma/PingUseCase.java
package com.qomolangma;

import com.qomolangma.frameworks.application.core.UseCase;
import com.qomolangma.frameworks.valueobjects.payload.RequestPayload;
import com.qomolangma.frameworks.valueobjects.payload.ResponsePayload;
import com.qomolangma.frameworks.valueobjects.payload.StringFieldType;

import static com.qomolangma.frameworks.valueobjects.payload.ResponsePayload.response;

@UseCase
public class PingUseCase {
    public ResponsePayload execute(RequestPayload payload) {
        return response().append("ping", payload.get("data", StringFieldType.asString()));
    }
}

File: src/test/kotlin/com/qomolangma/infra/PingUseCase.kt
package com.qomolangma.infra

import com.qomolangma.frameworks.application.core.UseCase
import com.qomolangma.frameworks.payload.core.Payload
import com.qomolangma.frameworks.security.core.Authorities
import javax.annotation.Resource

@UseCase(requiredAuthorities = [Authorities.AUTHENTICATED])
class PingUseCase {
    @Resource
    private val ping: Ping? = null

    fun execute(payload: Payload): Payload {
        return ping!!.pong(payload)
    }
}
